test/e2e/serial/tests/configuration.go

```diff
@@ -143,245 +140,100 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
-					Expect(deploy.WaitForMCPsCondition(fxt.Client, context.TODO(), machineconfigv1.MachineConfigPoolUpdating, initialMcpInfo.mcpObj)).To(Succeed())
-					waitForMcpUpdate(fxt.Client, context.TODO(), MachineConfig, time.Now().String(), initialMcpInfo)
-				}
 			}()
+			// keep the mcp with zero machine count intentionally to avoid reboots
 
-			//so far 0 machine count for mcp-test -> no nodes -> no updates status -> empty status
-			var updatedNewMcp machineconfigv1.MachineConfigPool
-			Expect(fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(mcp), &updatedNewMcp)).To(Succeed())
-
-			newMcpInfo := mcpInfo{
-				mcpObj:        &updatedNewMcp,
-				initialConfig: updatedNewMcp.Status.Configuration.Name,
-				sampleNode:    targetedNode,
+			testNG := nropv1.NodeGroup{
+				MachineConfigPoolSelector: &metav1.LabelSelector{
+					MatchLabels: testMCP.Labels,
+				},
 			}
-			klog.Infof("new mcp info: %s", newMcpInfo.ToString())
-
-			By(fmt.Sprintf("Label node %q with %q", targetedNode.Name, getLabelRoleMCPTest()))
-			unlabelFunc, err := labelNode(fxt.Client, getLabelRoleMCPTest(), targetedNode.Name)
-			Expect(err).ToNot(HaveOccurred())
-
-			defer func() {
-				By(fmt.Sprintf("CLEANUP: restore initial labels of node %q with %q", targetedNode.Name, getLabelRoleWorker()))
-				var updatedMcp machineconfigv1.MachineConfigPool
-				Expect(fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(initialMcpInfo.mcpObj), &updatedMcp)).To(Succeed())
-				initialMcpInfo.initialConfig = updatedMcp.Status.Configuration.Name
-
-				err = unlabelFunc()
-				Expect(err).ToNot(HaveOccurred())
-
-				Expect(deploy.WaitForMCPsCondition(fxt.Client, context.TODO(), machineconfigv1.MachineConfigPoolUpdating, initialMcpInfo.mcpObj)).To(Succeed())
-				// node reboot will be applied to the target node because it will belong now back to worker mcp which has specific KC/PP applied; NROP setting here will not affect here because the worker already has worker label; this is a sign we have an issue in NROP: one node group can init RTE on nodes from different MCPs! see bug from one customer back then
-				waitForMcpUpdate(fxt.Client, context.TODO(), MachineCount, time.Now().String(), initialMcpInfo)
-			}()
-
-			waitForMcpUpdate(fxt.Client, context.TODO(), MachineConfig, time.Now().String(), newMcpInfo)
 
-			By(fmt.Sprintf("modifying the NUMAResourcesOperator nodeGroups field to match new mcp: %q labels %q", mcp.Name, mcp.Labels))
+			By(fmt.Sprintf("modifying the NUMAResourcesOperator nodeGroups field to include new group: %q labels %q", testMCP.Name, testMCP.Labels))
+			newLogLevel := operatorv1.Trace
 			Eventually(func(g Gomega) {
 				// we need that for the current ResourceVersion
-				err := fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(initialNroOperObj), nroOperObj)
-				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(fxt.Client.Get(ctx, client.ObjectKeyFromObject(initialNroOperObj), &nroOperObj)).To(Succeed())
```

Comment:
it would be better to use a local variable to perform the update, but this also works

Reply:
yeah this didn't show any issue and the idea was to reuse the vars as much as relevant

Reply:
reusing vars is almost always a bad idea though

Reply:
I agree, in this case though the reuse is completely overriding the old value so no risk in using old data or something like that